File: ESTRUTURAS/1281/main.js
function getProducts() {
    let products = {};
    let m = parseInt(lines.shift());
    while (m--) {
        let [product, price] = lines.shift().split(' ');
        products[product] = parseFloat(price);
    }
    return products;
}

function getTotal(products) {
    let total = 0;
    let p = parseInt(lines.shift());
    while (p--) {
        let [product, qty] = lines.shift().split(' ');
        total += products[product] * parseInt(qty);
    }
    return total;
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').trim().split('\n');
let n = parseInt(lines.shift());
while (n--) {
    let products = getProducts();
    let total = getTotal(products);
    console.log(`R$ ${total.toFixed(2)}`);
}

File: ESTRUTURAS/1281/main.py
def get_products():
    products = {}
    m = int(input())
    for _ in range(m):
        product, price = input().split()
        products[product] = float(price)
    return products


def get_total(products):
    total = 0
    p = int(input())
    for _ in range(p):
        product, qty = input().split()
        total += products[product] * int(qty)
    return total


n = int(input())
for _ in range(n):
    products = get_products()
    total = get_total(products)
    print(f'R$ {total:.2f}')
